### PR TITLE
Save call

### PR DESCRIPTION
**Pull request by make-it-git**
## Overview

| Q                 | A
| ----------------- | ---
| Bugfix?           | no
| New feature?      | yes
| Breaking Changes? | no
| Fixed issues      | #

## Notes

May be we should make 'id' field nullable for all models.
When we send request to SageOne without 'ID' field it returns new entity with new 'ID'

Notify the following people: @darrynten

### CODE REVIEW

File: tests/SageOne/Models/AccountOpeningBalanceModelTest.php [hunk 1]
```diff
@@ -132,6 +140,14 @@ public function testSaveBadResponsePropertyHasInvalidType()
         $this->expectExceptionMessage('Model "AccountOpeningBalance" Received namespaced class "DarrynTen\SageOne\Models\double" when defined type is "boolean" Property is referencing an undefined, non-primitive class');
         $this->expectExceptionCode(10110);
 
+        $model->loadResult(
+            json_decode(
+                file_get_contents(
+                    __DIR__ . '/../../mocks/AccountOpeningBalance//POST_AccountOpeningBalance_Save_REQ.json'
```
Comment (darrynten) [inline tests/SageOne/Models/AccountOpeningBalanceModelTest.php]: There's a double `//` here

**Review by darrynten — CHANGES_REQUESTED** [review]
Small typo :)

**Comment by darrynten** [comment]
> Maybe we should make 'id' field nullable for all models.
> When we send request to SageOne without 'ID' field it returns new entity with new 'ID'

@make-it-git @fergusdixon I think this is a good idea

This kind of turns it into a create and an update call though, what do you think about that?

Additionally, the xero sister project of this one has the concept of create and update, I'm wondering if we should go one step further and split this out into a create and an update call?

But then again perhaps something as simple as making them all nullable would suffice.

Thoughts?

**Comment by make-it-git** [comment]
I think there is no requirement for separate update() call.

save() anyway: returns new model or updates current

**Review by darrynten — APPROVED** [review]

**Pull request merged**